docs/enhancements/soft-install-timeout.md

```diff
@@ -0,0 +1,172 @@
+| Rebooting                  | host|                                  stage |    Assisted service |                 HOST_STAGE_REBOOTING_TIMEOUT |                       40m |                                      Move pending user action |
+| Wait for nodes             | installation (cluster)| controller | Assisted controller | hard coded |                       10h |                                       Abort waiting for nodes |
+| Wait for finalizing        | installation (cluster)| controller | Assisted controller | hard coded |                       10h | Don't perform post install + don't send complete installation |
+| Wait for cluster operators | installation (cluster)| controller | Assisted controller | hard coded |                       10h |  Don't perform rest of controller operations | Only CVO and console |
+| Add router CA              | installation (cluster)| controller | Assisted controller | hard coded |                       70m | Don't perform rest of controller operations |
+| Wait for OLM operators     | installation (cluster)| controller | Assisted controller || calculated from operators | | Don't perform rest of controller operations |
+| Apply manifests            | installation (cluster)| controller | Assisted controller | hard coded |                       10m |  Don't perform rest of controller operations |
+| Wait for OLM operators CSV | installation (cluster)| controller | Assisted controller | | calculated from operators | Don't perform rest of controller operations |
+| Send complete installation | installation (cluster)| controller | Assisted controller | hard coded | 30m |  Don't perform rest of controller operations |
+
+
+There are 2 flows for completing installation.  One is managed by controller, and one by assisted service.
+
+In the flow managed by the assisted service, the following steps must be completed:
+
+- kubeconfig must be uploaded
+- cluster operators are successful
+- Monitored operators are either successful or failed
+
+In the flow managed by the controller all steps must be completed (in the table above).  The last step is a 
+a notification to complete installation.
+
+### Suggested changes
+- Only assisted service should notify installation completion
+- All steps by controller will be known as stages in the assisted service.  This will enable the service to manage 
+them in a similar way that host stages are managed.
+- Controller will not stop activity due to timeout.
+- Assisted service should be able to terminate the controller.
+- All existing timeouts except installation timeout and rebooting timeout should be treated as soft timeouts (i.e notification only)
+- In case of some optional operation (i.e managed operators), the user will be able to decide to skip it and set the cluster as completed.
+
+### Open Questions
+
+- Do we also soften the 24h `INSTALLATION_TIMEOUT` timeout?
+
+- Should we collect must gather logs when the installation timeout expires, even
```

Comment:
when triage a cluster you should get the latest state no? 
Or you suspect that the log size will increase?

Reply:
I think we should gather the logs, even if the cluster is not marked as failed

Reply:
what about the logs size? 
if possible lets check how the size will be affected, currently we have docs that estimate the size and require the users to create a PV matching that size, i know that the behavior will be disabled for ACM but in case we enable it it's good to know what are the limits